### PR TITLE
Add a ‘log’ window

### PR DESCRIPTION
**Pull request by swashcap**
* **Problem:** Identifying bugs and their steps is difficult with a production-build application (see #152).
* **Solution:** Add a logging window.
* **Testing:**
  1. Blow away your logs (`rm -rf ~/Library/Logs/coinstac` on OS X)
  2. Boot the UI
  3. Check the “Logs” menu item.
  4. packing-up
  5. 🎉

### CODE REVIEW

**Comment by swashcap** [comment]
This uses [tail](https://www.npmjs.com/package/tail) to essentially `tail -F` the log file.

File: packages/coinstac-ui/app/render/log.html [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        color: red;
+      }
+    </style>
+  </head>
+  <body>
+    <h1>COINSTAC Log</h1>
+
+    <pre id="output"></pre>
+
+    <script src="log.js"></script>
```
Comment (swashcap) [inline packages/coinstac-ui/app/render/log.html:112]: Some virtue in divorcing the logger from the render UI build process, I think. We can always add it as a new entry in our Webpack bundles, but it’s simple enough to stand on its own for now.

File: packages/coinstac-ui/app/main/utils/boot/configure-browser-window.js [hunk 2]
```diff
@@ -8,6 +8,9 @@
 
 const app = require('ampersand-app');
 const electron = require('electron');
+const electronDefaultMenu = require('electron-default-menu');
```
Comment (swashcap) [inline packages/coinstac-ui/app/main/utils/boot/configure-browser-window.js:11]: Electron’s `Menu`’s kinda funny: it’s [apparently read-only](https://github.com/electron/electron/issues/8928#issuecomment-286664574)? This module helps reconstruct the default menu; our “Logs” menu item is added below.

File: packages/coinstac-ui/app/main/utils/boot/configure-browser-window.js [hunk 3]
```diff
@@ -38,6 +42,38 @@ function createWindow() {
...
     app.mainWindow = null;
   });
+
+  // Set app's menu
+  menu[0].submenu.splice(1, 0,
+    {
+      type: 'separator',
+    },
+    {
+      label: 'Logs',
```
Comment (swashcap) [inline packages/coinstac-ui/app/main/utils/boot/configure-browser-window.js:52]: :wave: Hello, logs!

File: packages/coinstac-ui/app/render/log.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  .then((config) => {
+    const target = path.join(
+      os.homedir(),
+      config.get('logLocations')[os.platform()],
+      config.get('logFile')
+    );
+
+    console.log(`Reading log: ${target}`); // eslint-disable-line no-console
+
+    return Promise.all([target, readLastLines.read(target, 50)]);
```
Comment (swashcap) [inline packages/coinstac-ui/app/render/log.js]: [tail](https://www.npmjs.com/package/tail) doesn’t support a `-n`-style number flag: it either reads the entire log file – :scream: – or watches for new lines. [read-last-lines](https://www.npmjs.com/package/read-last-lines) dumps the last 50 lines here.

**Comment by swashcap** [comment]
@rssk This looks good to me. Can you confirm it works on Windows?

**Review by rssk — APPROVED** [review]
So much log

**Pull request merged**